public/app/plugins/panel/flamegraph/components/FlameGraph/FlameGraphMetadata.tsx

```diff
@@ -1,53 +1,113 @@
+import { FilterPill, Icon, useStyles2 } from '@grafana/ui';
 
-import { Metadata } from '../types';
+import { ClickedItemData } from '../types';
 
 import { FlameGraphDataContainer } from './dataTransform';
 
 type Props = {
   data: FlameGraphDataContainer;
   totalTicks: number;
-  focusedItemIndex?: number;
+  onFocusPillClick: () => void;
+  onSandwichPillClick: () => void;
+  focusedItem?: ClickedItemData;
+  sandwichedLabel?: string;
 };
 
-const FlameGraphMetadata = React.memo(({ data, focusedItemIndex, totalTicks }: Props) => {
-  const styles = useStyles2(getStyles);
-  const metadata = getMetadata(data, focusedItemIndex || 0, totalTicks);
-  const metadataText = `${metadata?.unitValue} (${metadata?.percentValue}%) of ${metadata?.samples} total samples (${metadata?.unitTitle})`;
-  return <>{<div className={styles.metadata}>{metadataText}</div>}</>;
-});
+const FlameGraphMetadata = React.memo(
+  ({ data, focusedItem, totalTicks, sandwichedLabel, onFocusPillClick, onSandwichPillClick }: Props) => {
+    const styles = useStyles2(getStyles);
+    const parts: ReactNode[] = [];
+    const ticksVal = getValueFormat('short')(totalTicks);
+
+    const displayValue = data.valueDisplayProcessor(totalTicks);
+    let unitValue = displayValue.text + displayValue.suffix;
+    const unitTitle = data.getUnitTitle();
+    if (unitTitle === 'Count') {
+      if (!displayValue.suffix) {
+        // Makes sure we don't show 123undefined or something like that if suffix isn't defined
+        unitValue = displayValue.text;
+      }
+    }
 
-export const getMetadata = (data: FlameGraphDataContainer, itemIndex: number, totalTicks: number): Metadata => {
-  const displayValue = data.getValueDisplay(itemIndex);
-  const percentValue = Math.round(10000 * (displayValue.numeric / totalTicks)) / 100;
-  let unitValue = displayValue.text + displayValue.suffix;
+    parts.push(
+      <FilterPill
+        key={'default'}
+        className={styles.pill}
+        label={`${unitValue} | ${ticksVal.text}${ticksVal.suffix} samples (${unitTitle})`}
+        selected={false}
+        onClick={() => {}}
+      />
+    );
 
-  const unitTitle = data.getUnitTitle();
-  if (unitTitle === 'Count') {
-    if (!displayValue.suffix) {
-      // Makes sure we don't show 123undefined or something like that if suffix isn't defined
-      unitValue = displayValue.text;
+    if (sandwichedLabel) {
+      parts.push(
+        <span key={'sandwich'}>
+          {' > '}
+          <FilterPill
+            className={styles.pill}
+            label={
+              <>
+                <Icon containerClassName={styles.pillIcon} size={'sm'} name={'gf-show-context'} />{' '}
+                {sandwichedLabel.substring(sandwichedLabel.lastIndexOf('/') + 1)}
+              </>
+            }
+            selected={false}
+            tooltip={'Remove sandwich view'}
+            onClick={onSandwichPillClick}
+            aria-label={'Remove sandwich view'}
```

Comment:
Is whatever `sandwichedLabel` is relevent to what's being removed? Why isn't the contents of the FilterPill not just "Remove sandwich view"?

The aria-label _removes_ is removing that context.

Reply:
Yeah, after seeing the video, i think this aria-label is harmful as it's _removing_ content from the page. I'm not sure what the better option is though - "Remove {sandwichedLabel}" is probably the least worst?

The struggle is that this button is used for two purposes:
 - to inform the user of the current state (that's it' in a sandwhich view of this type)
 - to navigate out of this drill down view

Reply:
I feel like at that point I can probably just use a custom component here and add a small `X` button in the corner. I thought I would save some work by reusing the pills (and seem like it would be nicely consistent as I thought about this elements as sort of filters) but that did not seem the case right now :D

Reply:
I don't (necessarily) think you've used a wrong component here, but just that the aria-label is obscuring the visible content of the label. An explicit "Remove" button might result in UI that's better for everyone to understand, but my intent wasn't to suggest a redesign 😅